tgz.go

```diff
@@ -11,38 +11,42 @@ import (
 	"time"
 )
 
-type Tgz struct {
+type Archive struct {
 	Path      string
 	tgzFile   *os.File
 	tarWriter *tar.Writer
 	gzWriter  *gzip.Writer
 	finished  bool
 }
 
-func New(path string) (error, *Tgz) {
-	tgz := Tgz{Path: path}
+func New(path string) (*Archive, error) {
+	tgz := Archive{Path: path}
 	var err error
-	err, tgz.tgzFile = tgz.getTarFile()
+	tgz.tgzFile, err = tgz.getTarFile()
 	if err != nil {
-		return err, nil
+		return nil, err
 	}
 
 	tgz.gzWriter = gzip.NewWriter(tgz.tgzFile)
 	tgz.tarWriter = tar.NewWriter(tgz.gzWriter)
 	tgz.finished = false
 
-	return nil, &tgz
+	return &tgz, nil
 }
 
-func (tgz *Tgz) AddFileByPath(srcFile string, dest string) error {
+func (tgz *Archive) AddFileByBuffer(b *bytes.Buffer, dest string) error {
+	return tgz.AddFileByContent(b.Bytes(), dest)
+}
+
+func (tgz *Archive) AddFileByPath(srcFile string, dest string) error {
 	if src, err := ioutil.ReadFile(srcFile); err == nil {
 		return tgz.AddFileByContent(src, dest)
 	} else {
 		return err
 	}
 }
 
-func (tgz *Tgz) AddFileByContent(src []byte, dest string) error {
+func (tgz *Archive) AddFileByContent(src []byte, dest string) error {
 	if tgz.finished == true {
```

Comment:
hmm not sure this is needed. If you try and write to a closed writer it will error or maybe even panic